std/assembly/internal/itoa.ts

```diff
@@ -157,15 +155,13 @@ function utoa64_lut(buffer: usize, num: u64, offset: u32): void {
     store<u64>(buffer + (offset << 1), digits1 | (digits2 << 32), STRING_HEADER_SIZE);
   }
 
-  r = <u32>num;
```

Comment:
I think this should leave "as is" according to generated wast code. Not sure `i64.ne` interpret same as `i32.ne`. Last one should be faster.

Reply:
The difference there is that previously the value is wrapped to an i32 before the comparison, which always truncates the upper 32 bits, while afterwards the assumption is that the upper bits do not matter, even if set, while avoiding the possibly unnecessary wrap in WASM64.

Reply:
Yes, I see. Hmm, I'll check which is better in WAS. Everything other looks great)

Reply:
At this place, below is a check for non-zero at `if (num)`, so before it is a check whether the low 32-bits are non-zero, while after it is a check that the full 64-bits are non-zero, hmm.

Reply:
Appears that due to the loop above, `num` is guaranteed to be less than `100000000` (also means: less than 32-bits), so that should just work.

Reply:
Yah, may be we can drop `if` statement at all. case when `num` will be zero is really improbable. So just call `utoa32_lut(buffer, <u32>num, offset)` without checking

Reply:
I know about redundant `num = N>0 * 100000000` but I expect `countDecimal` propably using in future or externally in implement this in general way

Reply:
`utoa32_lut` pass inside `utoa64_lut` always execute because when we have `num < 2^32-1` we just call `utoa32_lut` with downcasting to `i32`